src/common/url.cpp

```diff
@@ -4,19 +4,33 @@
 
 #include <common/url.h>
 
-#include <event2/http.h>
-
-#include <cstdlib>
 #include <string>
+#include <stdexcept>
+#include <charconv>
 
-std::string urlDecode(const std::string &urlEncoded) {
+std::string urlDecode(const std::string& urlEncoded) {
```

Comment:
We didn't use [plus decoding](https://github.com/libevent/libevent/blob/e0a4574ba2cbcdb64bb2b593e72be7f7f4010746/http.c#L3505) with `evhttp_uridecode`, so this implementation keeps functionality the same. Since `urlDecode` is a generic utility function, I think there should be a docstring that highlights that this implementation doesn't decode `+` to ` ` in case it's used for other purposes in the future.

Reply:
Added a short docstring.